conan/tools/cmake/presets.py

```diff
@@ -28,11 +28,11 @@ def generate(conanfile, toolchain_file, generator, cache_variables, preset_prefi
             if "CMAKE_SH" not in cache_variables:
                 cache_variables["CMAKE_SH"] = "CMAKE_SH-NOTFOUND"
 
-            cmake_make_program = conanfile.conf.get("tools.gnu:make_program",
-                                                    default=cache_variables.get("CMAKE_MAKE_PROGRAM"))
-            if cmake_make_program:
-                cmake_make_program = cmake_make_program.replace("\\", "/")
-                cache_variables["CMAKE_MAKE_PROGRAM"] = cmake_make_program
+        cmake_make_program = conanfile.conf.get("tools.gnu:make_program",
+                                                default=cache_variables.get("CMAKE_MAKE_PROGRAM"))
+        if cmake_make_program:
+            cmake_make_program = cmake_make_program.replace("\\", "/")
+            cache_variables["CMAKE_MAKE_PROGRAM"] = cmake_make_program
```

Comment:
I know this was already there, and not strictly related to this PR. But why is it a cache variable (presets) and not a toolchain one? Because it means that this is an extra things that users not using presets should remember to pass in the command line too. It would be worth checking if this works in the toolchain (probably will fail, and this is the reason it is a cache variable)